projects/rates.go

```diff
@@ -0,0 +1,1303 @@
+type RateProjectAndUsersUpdateRequest struct {
+	// Path contains the path parameters for the request.
+	Path RateProjectAndUsersUpdateRequestPath `json:"-"`
+
+	// ProjectRate is the new rate for the project.
+	ProjectRate int64 `json:"projectRate"`
+
+	// UserRates contains the user rates to set as exceptions.
+	UserRates []ProjectUserRateRequest `json:"userRates,omitempty"`
+}
+
+// NewRateProjectAndUsersUpdateRequest creates a new RateProjectAndUsersUpdateRequest.
+func NewRateProjectAndUsersUpdateRequest(projectID int64, projectRate int64) RateProjectAndUsersUpdateRequest {
+	return RateProjectAndUsersUpdateRequest{
+		Path: RateProjectAndUsersUpdateRequestPath{
+			ProjectID: projectID,
+		},
+		ProjectRate: projectRate,
+	}
+}
+
+// HTTPRequest creates an HTTP request for the RateProjectAndUsersUpdateRequest.
+func (r RateProjectAndUsersUpdateRequest) HTTPRequest(ctx context.Context, server string) (*http.Request, error) {
+	uri := fmt.Sprintf("%s/projects/api/v3/rates/projects/%d/actions/update", server, r.Path.ProjectID)
+
+	var body bytes.Buffer
+	if err := json.NewEncoder(&body).Encode(r); err != nil {
+		return nil, fmt.Errorf("failed to encode update project and users rate request: %w", err)
+	}
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodPut, uri, &body)
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Set("Content-Type", "application/json")
+
+	return req, nil
+}
+
+// RateProjectAndUsersUpdateResponse represents the response for updating a project rate and user rates.
+type RateProjectAndUsersUpdateResponse struct{}
+
+// HandleHTTPResponse handles the HTTP response for the RateProjectAndUsersUpdateResponse.
+func (r *RateProjectAndUsersUpdateResponse) HandleHTTPResponse(resp *http.Response) error {
+	if resp.StatusCode != http.StatusNoContent {
+		return twapi.NewHTTPError(resp, "failed to update project and users rates")
+	}
+	return nil
+}
+
+// RateProjectAndUsersUpdate updates a project rate and user rates using the provided request and returns the response.
+func RateProjectAndUsersUpdate(
+	ctx context.Context,
+	engine *twapi.Engine,
+	req RateProjectAndUsersUpdateRequest,
+) (*RateProjectAndUsersUpdateResponse, error) {
+	return twapi.Execute[RateProjectAndUsersUpdateRequest, *RateProjectAndUsersUpdateResponse](ctx, engine, req)
+}
+
+// RateProjectUserListRequestPath contains the path parameters for listing project user rates.
+type RateProjectUserListRequestPath struct {
+	// ProjectID is the unique identifier of the project.
+	ProjectID int64
+}
+
+// RateProjectUserListRequestFilters contains the filters for listing project user rates.
+type RateProjectUserListRequestFilters struct {
+	// SearchTerm is an optional search term to filter by first name or last name.
+	SearchTerm string
+
+	// OrderBy specifies the ordering of results.
+	OrderBy string
+
+	// OrderMode specifies the order direction (asc, desc).
+	OrderMode string
```

Comment:
This could also benefit from ENUMs instead of `string`.

```go
// In the twapi top-level:

type OrderMode string

const (
  OrderModeAscending  OrderMode = "asc"
  OrderModeDescending OrderMode = "desc"
)

// In this file

type RateProjectUserListOrderBy string

const (
  // all sorting options for this endpoint
)
```

```suggestion
	// OrderBy specifies the ordering of results.
	OrderBy RateProjectUserListOrderBy

	// OrderMode specifies the order direction (asc, desc).
	OrderMode twapi.OrderMode
```

This would also help the SDK user avoid searching for documentation to find the available options.